apps/api/src/billing/services/chain-error/chain-error.service.spec.ts

```diff
@@ -0,0 +1,129 @@
+import type { BalanceHttpService } from "@akashnetwork/http-sdk";
+import type { EncodeObject } from "@cosmjs/proto-signing";
+import type { MockProxy } from "jest-mock-extended";
+import { mock } from "jest-mock-extended";
+
+import type { Wallet } from "@src/billing/lib/wallet/wallet";
+import type { BillingConfigService } from "@src/billing/services/billing-config/billing-config.service";
+import { ChainErrorService } from "./chain-error.service";
+
+const USDC_IBC_DENOMS = {
+  mainnetId: "ibc/170C677610AC31DF0904FFE09CD3B5C657492170E7E52372E48756B71E56F2F1",
+  sandboxId: "ibc/12C6A0C374171B595A0A9E18B83FA09D295FB1F2D8C6DAA3AC28683471752D84"
+};
+
+describe(ChainErrorService.name, () => {
+  function setup(): {
+    balanceHttpService: MockProxy<BalanceHttpService>;
+    billingConfigService: MockProxy<BillingConfigService>;
+    masterWallet: MockProxy<Wallet>;
+    service: ChainErrorService;
+  } {
+    const balanceHttpService = mock<BalanceHttpService>();
+    const billingConfigService = mock<BillingConfigService>();
+    const masterWallet = mock<Wallet>();
+
+    // Stub configuration values
+    billingConfigService.get.mockImplementation((key: string) => {
+      if (key === "USDC_IBC_DENOMS") return USDC_IBC_DENOMS;
+      if (key === "DEPLOYMENT_GRANT_DENOM") return "uakt";
+      return undefined as any;
+    });
+
+    masterWallet.getFirstAddress.mockResolvedValue("test-address");
+
+    const service = new ChainErrorService(balanceHttpService, billingConfigService, masterWallet);
+
+    return { balanceHttpService, billingConfigService, masterWallet, service };
+  }
+
+  describe("parseInsufficientFundsErrorMessage", () => {
+    it("parses uakt correctly", () => {
+      const { service } = setup();
+      const msg = "42uakt is smaller than 100uakt";
+      expect((service as any).parseInsufficientFundsErrorMessage(msg)).toEqual({
```

Comment:
could you pls re-write the test to cover public methods only. no need to do casting here. private methods can change more frequently and public tend to keep interface more.

Reply:
Sure, done